src/any_agent/frameworks/google.py

```diff
@@ -20,53 +19,39 @@
 
 
 class GoogleAgent(AnyAgent):
-    """Google agent implementation that handles both loading and running."""
+    """Google ADK agent implementation that handles both loading and running."""
 
     def __init__(
         self, config: AgentConfig, managed_agents: list[AgentConfig] | None = None
     ):
-        if not adk_available:
-            msg = "You need to `pip install 'any-agent[google]'` to use this agent"
-            raise ImportError(msg)
         self.managed_agents = managed_agents
         self.config = config
         self._agent = None
-        self._mcp_servers = None
-        self._managed_mcp_servers = None
+        self._mcp_servers = []
+        self.framework = AgentFramework.GOOGLE
```

Comment:
Adding this change seem like a good oportunity to introduce Discriminated Unions with Pydantic, I can draft a PR after this one is merged.

That will simplify several places of the logic when creating framework-specific agents

Reply:
Cool yeah that would be good. We put Discriminated Unions into our Lumigator project codebase a few months ago and have been happy with that design pattern 👍